app/src/main/java/org/schabi/newpipe/MainActivity.java

```diff
@@ -449,6 +449,16 @@ protected void onResume() {
             sharedPreferences.edit().putBoolean(Constants.KEY_MAIN_PAGE_CHANGE, false).apply();
             NavigationHelper.openMainActivity(this);
         }
+
+        if (sharedPreferences.getBoolean(Constants.KEY_ENABLE_WATCH_HISTORY, true)) {
+            if (DEBUG) Log.d(TAG, "do not show History-menu as its disabled in settings");
+            drawerItems.getMenu().findItem(ITEM_ID_HISTORY).setVisible(true);
+        }
+
+        if (!sharedPreferences.getBoolean(Constants.KEY_ENABLE_WATCH_HISTORY, true)) {
```

Comment:
This check is just a duplicate from above, a variable here would make much more sense:

```java
final boolean isHistoryEnabled = sharedPreferences.getBoolean(
        getString(R.string.enable_watch_history_key), true);
drawerItems.getMenu().findItem(ITEM_ID_HISTORY).setVisible(isHistoryEnabled);
```